nilearn/maskers/multi_nifti_masker.py

```diff
@@ -240,8 +240,15 @@ def fit(self, imgs=None, y=None):
 
         self._reporting_data = None
         if self.reports:  # save inputs for reporting
-            imgs = imgs[0] if isinstance(imgs, list) else imgs
-            self._reporting_data = {"images": imgs, "mask": self.mask_img_}
+            self._reporting_data = {"mask": self.mask_img_, "images": imgs}
+            if imgs is not None:
+                imgs = imgs[0] if isinstance(imgs, list) else imgs
+                dim = image.load_img(imgs).shape
+                if len(dim) == 4:
+                    # compute middle image from 4D series for plotting
+                    imgs = image.index_img(imgs, dim[-1] // 2)
```

Comment:
Can this be factored in a function ? I guess this pattern already exists in the code base ?

Reply:
Yes probably, let me look into it